py/frequenz/client/electricity_trading/_types.py

```diff
@@ -233,28 +233,44 @@ def to_pb(self) -> delivery_area_pb2.DeliveryArea:
         return delivery_area_pb2.DeliveryArea(code=self.code, code_type=self.code_type)
 
 
-class DeliveryDuration(enum.Enum):
+class DeliveryDuration:
     """
     Specifies the time increment, in minutes, used for electricity deliveries and trading.
 
     These durations serve as the basis for defining the delivery period in contracts, and they
     dictate how energy is scheduled and delivered to meet contractual obligations.
     """
 
-    UNSPECIFIED = delivery_duration_pb2.DeliveryDuration.DELIVERY_DURATION_UNSPECIFIED
-    """Default value, indicates that the duration is unspecified."""
+    def __init__(self, duration: timedelta):
+        """
+        Initialize a DeliveryDuration object.
+
+        Args:
+            duration: Duration of the delivery period.
+
+        Raises:
+            ValueError: If the duration is not 5, 15, 30, or 60 minutes.
+        """
+        if duration == timedelta(minutes=5):
+            self.duration = delivery_duration_pb2.DeliveryDuration.DELIVERY_DURATION_5
 
-    MINUTES_5 = delivery_duration_pb2.DeliveryDuration.DELIVERY_DURATION_5
-    """5-minute duration."""
+        elif duration == timedelta(minutes=15):
+            self.duration = delivery_duration_pb2.DeliveryDuration.DELIVERY_DURATION_15
 
-    MINUTES_15 = delivery_duration_pb2.DeliveryDuration.DELIVERY_DURATION_15
-    """15-minute contract duration."""
+        elif duration == timedelta(minutes=30):
+            self.duration = delivery_duration_pb2.DeliveryDuration.DELIVERY_DURATION_30
 
-    MINUTES_30 = delivery_duration_pb2.DeliveryDuration.DELIVERY_DURATION_30
-    """30-minute contract duration."""
+        elif duration == timedelta(minutes=60):
+            self.duration = delivery_duration_pb2.DeliveryDuration.DELIVERY_DURATION_60
 
-    MINUTES_60 = delivery_duration_pb2.DeliveryDuration.DELIVERY_DURATION_60
-    """1-hour contract duration."""
+        else:
+            # Default value, indicates that the duration is unspecified.
+            self.duration = (
+                delivery_duration_pb2.DeliveryDuration.DELIVERY_DURATION_UNSPECIFIED
+            )
+            raise ValueError(
+                "Invalid duration value. Duration must be 5, 15, 30, or 60 minutes."
+            )
```

Comment:
I think raising here somehow contradicts the default initialization. You might want to either set the default for `self.duration` or raise `ValueError` but probably not both in the initializer. Or I might be missing something...do you have any scenario where both might be needed?

Reply:
Nope you're right setting to the default makes more sense